File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/InAppLoginModule.java
package com.dashlane.dagger.singleton;

import android.content.Context;

import com.dashlane.inapplogin.InAppLoginByAccessibilityManager;
import com.dashlane.inapplogin.InAppLoginByAutoFillApiManager;
import com.dashlane.inapplogin.InAppLoginManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.android.qualifiers.ApplicationContext;

@Module
public class InAppLoginModule {

    @Provides
    @Singleton
    public InAppLoginManager provideInAppLoginManager(@ApplicationContext Context context) {
        return new InAppLoginManager(new InAppLoginByAccessibilityManager(context),
                                     InAppLoginByAutoFillApiManager.createIfPossible(context));
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/InAppLoginModule.kt
package com.dashlane.dagger.singleton

import android.content.Context
import com.dashlane.inapplogin.AutoFillApiManager.Companion.createIfPossible
import com.dashlane.inapplogin.InAppLoginManager
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object InAppLoginModule {
    @Provides
    @Singleton
    fun provideInAppLoginManager(@ApplicationContext context: Context) = InAppLoginManager(
        autoFillApiManager = createIfPossible(context)
    )
}